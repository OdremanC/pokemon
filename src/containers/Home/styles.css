@value colors: "../../assets/styles/variables.css";
@value yellow, blue, white from colors;

.home {
  margin: 20px;
  display: flex;
  justify-content: space-around;
  align-items: center;
  flex-wrap: wrap;
  min-height: 60vh;
}
