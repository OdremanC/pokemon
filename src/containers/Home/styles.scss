
.home {
  width: 100%;
  display: flex;
}

.main-container {
  width: 100%;
  height: 63vh;
  display: block;
  
}
