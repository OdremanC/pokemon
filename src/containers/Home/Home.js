import React, { useState, useEffect, Fragment} from 'react';
import { useTranslation } from "react-i18next";
import Header from '../../components/Header/Header';
import Footer from '../../components/Footer/Footer';
import Loader from '../../components/Loader/Loader';
import Paginator from '../../components/Paginator/Paginator';
import PokemonModel from '../../models/pockemonModel';
import List from '../../components/List/list';
import Details from '../../components/Details/Details';
import { getPokemonList, RESULT_LIMIT } from '../../utils/constants';
import './styles.css';

const Home = () => {
  const [offSet, setOffset] = useState(0)
  const [pokemonSelected, setPokemonSelected] = useState({})
  const [loading, setLoading] = useState(true);
  const [pokemonList, setPokemonList] = useState([]);
  const [totalPokemons, setTotalPokemons] = useState(0);
  const [pokemonDetails, setPokemonDetails] = useState({});
  const [modeList, setMode] = useState(true);
  const {t}= useTranslation();  
  const pageCount = Math.ceil(totalPokemons / RESULT_LIMIT);

  useEffect(() => {
    async function getPokemonsData() {
      setLoading(true);
      let data = await PokemonModel.getPokemons(offSet);
      setTotalPokemons(data.data.count);
      let list = getPokemonList(data.data.results);
      setPokemonList(list);
      setLoading(false);
    }
    getPokemonsData();
  }, [offSet]);

  useEffect(() => {
    async function getPokemonDetails() {
      if (pokemonSelected.url) {
        setLoading(true);
        let data = await PokemonModel.getPokemonDetail(pokemonSelected.url);
        setPokemonDetails(data.data);
        setMode(false);
        setLoading(false);
      } else {
        setMode(true);
      }
      
    }
    getPokemonDetails();
  }, [pokemonSelected]);

  return (
    <Fragment>
      <Loader loading={loading} t={t}/>
      <Header/>
      {
        modeList ?
        <List 
          data={pokemonList}
          selected={pokemonSelected}
          getSelectedPokemon={(url, id) => setPokemonSelected(url, id) }
        />
        :
        <Details 
          t={t}
          selected={pokemonSelected}
          detail={pokemonDetails}
          getSelectedPokemon={(url, id) => setPokemonSelected(url, id) }
        />

      }
    
      <Paginator 
        pageCount={pageCount} 
        limit={RESULT_LIMIT}
        t={t}
        callback={(value) => setOffset(value)}/>
      <Footer t={t} />
    </Fragment>
  );
}

export default Home;