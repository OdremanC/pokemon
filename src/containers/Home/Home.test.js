import React from 'react';
import Home from './Home';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

describe('<Home />', () => {

    it('matches snapshot', () => {
        const wrapper = shallow(<Home />);
        expect(wrapper).toMatchSnapshot();
    });
    
})
