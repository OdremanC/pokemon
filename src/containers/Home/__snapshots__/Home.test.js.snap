// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Header />,
        <List
          data={Array []}
          details={Object {}}
          getSelectedPokemon={[Function]}
          selected={Object {}}
          t={[Function]}
        />,
        <Paginator
          callback={[Function]}
          limit={5}
          pageCount={0}
          t={[Function]}
        />,
        <Footer
          t={[Function]}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "data": Array [],
          "details": Object {},
          "getSelectedPokemon": [Function],
          "selected": Object {},
          "t": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "callback": [Function],
          "limit": 5,
          "pageCount": 0,
          "t": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "t": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Header />,
          <List
            data={Array []}
            details={Object {}}
            getSelectedPokemon={[Function]}
            selected={Object {}}
            t={[Function]}
          />,
          <Paginator
            callback={[Function]}
            limit={5}
            pageCount={0}
            t={[Function]}
          />,
          <Footer
            t={[Function]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "data": Array [],
            "details": Object {},
            "getSelectedPokemon": [Function],
            "selected": Object {},
            "t": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "callback": [Function],
            "limit": 5,
            "pageCount": 0,
            "t": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "t": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
